syne_tune/optimizer/schedulers/single_fidelity_scheduler.py

```diff
@@ -0,0 +1,173 @@
+    :type searcher: str or
+        :class:`~syne_tune.optimizer.schedulers.searchers.BaseSearcher`
+    :param metric: Name of metric to optimize, key in results obtained via
+        ``on_trial_result``. For multi-objective schedulers, this can also be a
+        list
+    :type metric: str or List[str]
+    :param mode: "min" if ``metric`` is minimized, "max" if ``metric`` is
+        maximized, defaults to "min". This can also be a list if ``metric`` is
+        a list
+    :type mode: str or List[str], optional
+    :param points_to_evaluate: List of configurations to be evaluated
+        initially (in that order). Each config in the list
+        can be partially specified, or even be an empty dict. For each
+        hyperparameter not specified, the default value is determined using
+        a midpoint heuristic.
+        If not given, this is mapped to ``[dict()]``, a single default config
+        determined by the midpoint heuristic. If ``[]`` (empty list), no initial
+        configurations are specified.
+        Note: If ``searcher`` is of type :class:`BaseSearcher`,
+        ``points_to_evaluate`` must be set there.
+    :type points_to_evaluate: ``List[dict]``, optional
+    :param random_seed: Master random seed. Generators used in the
+        scheduler or searcher are seeded using :class:`RandomSeedGenerator`.
+        If not given, the master random seed is drawn at random here.
+    :type random_seed: int, optional
+    :param time_keeper: This will be used for timing here (see
+        ``_elapsed_time``). The time keeper has to be started at the beginning
+        of the experiment. If not given, we use a local time keeper here,
+        which is started with the first call to :meth:`_suggest`. Can also be set
+        after construction, with :meth:`set_time_keeper`.
+        Note: If you use
+        :class:`~syne_tune.backend.simulator_backend.SimulatorBackend`, you need
+        to pass its ``time_keeper`` here.
+    :type time_keeper: :class:`~syne_tune.backend.time_keeper.TimeKeeper`,
+        optional
+    """
+
+    def __init__(
+        self,
+        config_space: Dict[str, Any],
+        metric: str,
+        do_minimize: Optional[bool] = True,
+        searcher: Optional[Union[str, BaseSearcher]] = "random_search",
+        random_seed: int = None,
+        searcher_kwargs: dict = None,
+        **kwargs,
+    ):
+        super().__init__(random_seed=random_seed)
+
+        self.metric = metric
+        self.config_space = config_space
+        self.do_minimize = do_minimize
+        self.metric_op = 1 if self.do_minimize else -1
+
+        if isinstance(searcher, str):
+            if searcher_kwargs is None:
+                searcher_kwargs = {}
+
+            self.searcher = searcher_factory(searcher, config_space, **searcher_kwargs)
+        else:
+            self.searcher = searcher
+
+    def suggest(self, trial_id: int) -> Optional[TrialSuggestion]:
+
+        trial_id = str(trial_id)
```

Comment:
Can we remove this string conversion here? (and move it to the downstream classes if they need it)
It may not be feasible in this PR given all that needs to be done but then can we add a todo and put a task for it?